src/stravalib/tests/integration/test_client.py

```diff
@@ -539,6 +543,54 @@ def _call_and_assert():
         _call_and_assert()
 
 
+@pytest.mark.parametrize(
+    "sport_type, activity_type, expected_result, expected_exception",
+    (
+        ("TrailRun", None, "TrailRun", None),
+        ("funrun", None, None, ValueError),
+        # Did strava used to lower case types but sport_type is normal case?
+        (None, "Run", "run", None),
+        (None, "junoDog", None, ValueError),
+        # Case where we have both sport and activity type provided
+        # This should ONLY return Trail run. we'd want a test for two things
+        ("TrailRun", "Run", "TrailRun", None),
+        # Need to work on a method code check for this one - it should return
+        # a error i think for create activity?
+        (None, None, None, ValueError),
+    ),
+)
+def test_validate_activity_type(
```

Comment:
@jsamoocha i haven't rewritten the other two tests - YET. but i wondered if you could look at this test to let me know if i'm headed in the right direction. i wasn't sure how to handle all of the various cases and ended up with a few conditionals. any feedback on this is welcome. i will next work on removing the various case scenarios from create_activity and update_activity as discussed. thank you.